prow/plank/controller.go

```diff
@@ -167,41 +160,42 @@ func (c *Controller) Sync() error {
 	close(pjCh)
 
 	errCh := make(chan error, len(pjs))
+	reportCh := make(chan kube.ProwJob, len(pjs))
```

Comment:
This will work, but I'm worried that it's going to be a problem due to our tens of thousands of prowjobs. An alternative is to keep the reports channel as a member of the controller struct, write to it asynchronously with `go func(pj kube.ProwJob) { c.reports <- pj }()`, and then use something like `for v, ok <- c.reports; ok { report(v) }`. WDYT?